tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -618,4 +625,108 @@ def test_data_loss(self):
         # The SI cache should have segments in it.
         # Check if those segment files existed before retention
         self.logger.debug('After fetch')
-        self._check_si_cache_after_fetch(data_dir_segments)
\ No newline at end of file
+        self._check_si_cache_after_fetch(data_dir_segments)
+
+    # Test the SI cache limit
+    @cluster(num_nodes=4)
+    def test_si_cache_limit(self):
+        self.logger.info(f"Environment: {os.environ}")
+        if os.environ.get('BUILD_TYPE', None) == 'debug':
+            self.logger.info(
+                "Skipping test in debug mode (requires release build)")
+            return
+
+        # Remote write/read set at topic level
+        rpk = RpkTool(self.redpanda)
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.write', 'true')
+        rpk.alter_topic_config(self.topic, 'redpanda.remote.read', 'true')
+        rpk.alter_topic_config(self.topic, 'retention.bytes',
+                               str(self.segment_size))
+
+        # 100k messages of size 2**18
+        # is ~24GB of data.
+        msg_size = 2**18
+        msg_count = 100000
+
+        self._gen_manifests(msg_size, self._fgo_node)
+
+        producer = FranzGoVerifiableProducer(self.test_context, self.redpanda,
+                                             self.topic, msg_size, msg_count,
+                                             self._fgo_node)
+
+        # The SI cache is set to 5GB in size. Test the cache limit by
+        # running 8 readers that consume 1GB segments effectively forcing
+        # ~8GB of IO. The readers read from random offsets.
+        rand_consumer = FranzGoVerifiableRandomConsumer(
+            self.test_context, self.redpanda, self.topic, msg_size, 10, 8,
+            self._fgo_node)
+
+        producer.start(clean=False)
+        rand_consumer.start(clean=False)
+
+        producer.wait()
+        rand_consumer.shutdown()
```

Comment:
This reads like its shutting down the consumer as soon as the producer is done, rather than waiting for the intended number of reads (needed to fill the cache) to happen.